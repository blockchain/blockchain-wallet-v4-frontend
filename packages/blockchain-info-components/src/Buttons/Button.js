import React from 'react'
import { darken } from 'polished'
import styled from 'styled-components'

const BaseButton = styled.button.attrs((props) => ({
  type: props.type ? props.type : 'button'
}))`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: ${(props) => (props.fullwidth ? '100%' : props.width ? props.width : 'auto')};
  min-width: ${(props) => (props.small ? '86px' : props.width ? props.width : '140px')};
  height: ${(props) => (props.jumbo ? '56px' : props.small ? '32px' : props.height)};
  padding: ${(props) => (props.padding ? props.padding : '10px 15px')};
  margin: ${(props) => props.margin};
  box-sizing: border-box;
  user-select: none;
  text-align: center;
  text-decoration: none;
  vertical-align: middle;
  letter-spacing: normal;
  transition: all 0.2s ease-in-out;
  white-space: nowrap;
  line-height: 1;
  text-transform: ${(props) =>
    props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none'};
  font-family: 'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu,
    Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: ${(props) => (props.jumbo ? '16px' : props.size)};
  font-weight: ${(props) => (props.jumbo ? '600' : '600')};
  cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${(props) => (props.disabled ? 0.5 : 1)};
  color: ${(props) => props.theme[props.color]};
  background-color: ${(props) =>
    props.backgroundColor ? props.theme[props.backgroundColor] : 'transparent'};
  border-radius: ${(props) => (props.rounded ? '20px' : '8px')};
  border-style: solid;
  border-width: ${(props) => (props.rounded ? '2px' : '1px')};
  border-color: ${(props) => props.theme[props.borderColor]};

  &:hover {
    border-color: ${(props) =>
      props.disabled
        ? 'none'
        : props.hoverBorderColor
        ? props.theme[props.hoverBorderColor]
        : darken(0.1, props.theme[props.borderColor])};
    background-color: ${(props) =>
      props.disabled
        ? 'none'
        : props.backgroundColor
        ? darken(0.1, props.theme[props.backgroundColor])
        : 'transparent'};
  }

  &:focus {
    outline: 0;
  }
`

const selectColor = (nature, small) => {
  switch (nature) {
    case 'dark':
      return {
        backgroundColor: 'grey800',
        borderColor: 'grey800',
        color: 'white'
      }
    case 'dark-grey':
      return {
        backgroundColor: 'grey800',
        borderColor: 'grey800',
        color: 'white'
      }
    case 'empty':
      return {
        backgroundColor: 'white',
        borderColor: 'grey000',
        color: small ? 'blue600' : 'grey800',
        hoverBorderColor: 'white'
      }
    case 'empty-blue': {
      return {
        backgroundColor: 'white',
        borderColor: 'grey100',
        color: 'blue600',
        hoverBorderColor: 'blue600'
      }
    }
    case 'empty-red': {
      return {
        backgroundColor: 'white',
        borderColor: 'grey100',
        color: 'red600',
        hoverBorderColor: 'red600'
      }
    }
    case 'empty-secondary':
      return {
        backgroundColor: 'white',
        borderColor: 'blue600',
        color: 'blue600'
      }
    case 'gray':
      return {
        backgroundColor: 'grey500',
        borderColor: 'grey500',
        color: 'white'
      }
    case 'light':
      return {
        backgroundColor: 'white',
        borderColor: 'grey000',
        color: 'blue600'
      }
    case 'light-red':
      return {
        backgroundColor: 'white',
        borderColor: 'grey000',
        color: 'red400'
      }
    case 'exchangeTurquoise':
      return {
        backgroundColor: 'exchangeTurquoise',
        borderColor: 'exchangeTurquoise',
        color: 'exchangeNight'
      }
    case 'primary':
      return {
        backgroundColor: 'blue600',
        borderColor: 'blue600',
        color: 'white'
      }
    case 'purple':
      return {
        backgroundColor: 'purple',
        borderColor: 'purple',
        color: 'white'
      }
    case 'secondary':
      return {
        backgroundColor: 'blue900',
        borderColor: 'blue900',
        color: 'white'
      }
    case 'received':
      return {
        backgroundColor: 'received',
        borderColor: 'received',
        color: 'white'
      }
    case 'sent':
      return { backgroundColor: 'sent', borderColor: 'sent', color: 'white' }
    case 'success':
      return {
        backgroundColor: 'success',
        borderColor: 'success',
        color: 'white'
      }
    case 'transferred':
      return {
        backgroundColor: 'transferred',
        borderColor: 'transferred',
        color: 'white'
      }
    case 'warning':
      return { backgroundColor: 'error', borderColor: 'error', color: 'white' }
    case 'white-transparent':
      return {
        borderColor: 'white',
        color: 'white'
      }
    case 'white-blue':
      return {
        backgroundColor: 'white',
        borderColor: 'white',
        color: 'blue600'
      }
    case 'green':
      return {
        backgroundColor: 'green600',
        borderColor: 'green600',
        color: 'white'
      }
    case 'grey800':
      return {
        backgroundColor: 'grey800',
        borderColor: 'grey800',
        color: 'white'
      }
    default:
      return {
        backgroundColor: 'grey000',
        borderColor: 'grey200',
        color: 'grey800',
        hoverBorderColor: 'white'
      }
  }
}

const Button = (props) => {
  const { children, disabled, nature, small, ...rest } = props
  const { backgroundColor, borderColor, color, hoverBorderColor } = selectColor(nature, small)

  return (
    <BaseButton
      {...rest}
      small={small}
      disabled={disabled}
      color={color}
      backgroundColor={backgroundColor}
      borderColor={borderColor}
      hoverBorderColor={hoverBorderColor}
    >
      {children}
    </BaseButton>
  )
}

Button.defaultProps = {
  bold: false,
  capitalize: false,
  disabled: false,
  fullwidth: false,
  height: '40px',
  nature: 'empty',
  rounded: false,
  size: '14px',
  small: false,
  uppercase: false
}

export default Button
