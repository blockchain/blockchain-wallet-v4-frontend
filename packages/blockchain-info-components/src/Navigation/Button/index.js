import React from 'react'
import styled from 'styled-components'
import { Color } from '../../Colors'
import { trackButtonEvent } from '../Events'
import Link from '../Link'

const DefaultButton = styled('button', 'input[type=submit]')`
  display: inline-block;
  font-size: 1rem;
  font-weight: 600;
  padding: 0 1rem;
  border-radius: var(--smBorderRadius);
  text-align: center;
  cursor: pointer;
  outline: none;
  transition: all 0.3s ease;
  flex: 0 0 auto;

  border-radius: ${props =>
    props.big ? 'var(--lgBorderRadius)' : 'var(--smBorderRadius)'};

  height: ${props => (props.big ? '3.5rem' : '2.5rem')};

  font-size: ${props => (props.big ? '1.25rem' : '1rem')};

  padding: ${props => (props.big ? '0 2rem' : '0 1rem')};

  color: ${props => (props.textColor ? props.textColor : 'white')};

  border: ${props =>
    props.outline
      ? props.bgColor
        ? '2px solid ' + props.bgColor
        : `2px solid ${Color('brand-secondary')}`
      : 'transparent'};

  background-color: ${props =>
    props.outline
      ? 'transparent'
      : props.bgColor
      ? props.bgColor
      : Color('brand-secondary')};

  line-height: ${props => (props.outline ? '2rem' : '2.5rem')};

  &:hover {
    transform: scale(0.95);
    transition: all 0.3s ease;
  }

  a {
    color: inherit;
  }
`

const handleClick = props => {
  return () => {
    if (props.onClick) {
      props.onClick()
    }
    if (props.event) {
      trackButtonEvent(props.event)
    }
  }
}

const SecondaryButton = styled(DefaultButton)`
  background-color: rgba(1, 1, 1, 0.1);
  color: white;
`

const Button = props => {
  let { href, target, download, rel } = props
  if (href) {
    return (
      <Link
        href={href}
        target={target}
        download={download}
        rel={rel}
        onClick={handleClick(props)}
      >
        <DefaultButton
          big={props.big}
          bgColor={props.bgColor}
          outline={props.outline}
          textColor={props.textColor}
        >
          {props.children}
        </DefaultButton>
      </Link>
    )
  }
  return props.secondary ? (
    <SecondaryButton onClick={handleClick(props)}>
      {props.children}
    </SecondaryButton>
  ) : (
    <DefaultButton
      big={props.big}
      bgColor={props.bgColor}
      outline={props.outline}
      textColor={props.textColor}
      type={props.type}
      value={props.value}
      onClick={handleClick(props)}
    >
      {props.children}
    </DefaultButton>
  )
}

export default Button
