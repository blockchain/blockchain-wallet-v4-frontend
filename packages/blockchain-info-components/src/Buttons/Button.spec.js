import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Button from './Button'

describe('Button component', () => {
  it('default renders correctly', () => {
    const component = shallow(<Button />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
