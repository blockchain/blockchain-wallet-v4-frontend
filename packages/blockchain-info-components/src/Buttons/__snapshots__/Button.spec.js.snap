// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component default renders correctly 1`] = `
<styled.button
  backgroundColor="white"
  bold={false}
  borderColor="grey000"
  capitalize={false}
  color="grey800"
  disabled={false}
  fullwidth={false}
  height="40px"
  hoverBorderColor="white"
  rounded={false}
  size="14px"
  small={false}
  uppercase={false}
/>
`;
